framework/helpers/BaseJson.php

```diff
@@ -126,28 +127,23 @@ public static function decode($json, $asArray = true)
     /**
      * Handles [[encode()]] and [[decode()]] errors by throwing exceptions with the respective error message.
      *
-     * @param int $lastError error code from [json_last_error()](https://secure.php.net/manual/en/function.json-last-error.php).
+     * @param int $lastError error code from [json_last_error()](https://www.php.net/manual/en/function.json-last-error.php).
      * @throws InvalidArgumentException if there is any encoding/decoding error.
      * @since 2.0.6
      */
     protected static function handleJsonError($lastError)
     {
-        if ($lastError === JSON_ERROR_NONE) {
```

Comment:
Early return was alright, more readable.

Reply:
@samdark `return;` is `@return null`, not `@return void`

Reply:
They're equal: https://3v4l.org/J4hps

Reply:
@samdark I mean it's not "more readable"  than `if ($lastError !== JSON_ERROR_NONE) {...}`.

Reply:
It is. Early return reduces code nesting greatly.

Reply:
@samdark it's not important - short code & max nesting level is 2:
~~~php
protected static function handleJsonError($lastError)
{
    if ($lastError !== JSON_ERROR_NONE) {
        foreach (static::$jsonErrorMessages as $const => $message) {
            if (defined($const) && constant($const) === $lastError) {
                throw new InvalidArgumentException($message, $lastError);
            }
        }
        if (function_exists('json_last_error_msg')) {
            throw new InvalidArgumentException(json_last_error_msg(), $lastError);
        }
        throw new InvalidArgumentException('Unknown JSON encoding/decoding error.');
    }
}
~~~

Reply:
Compare with:

```php
protected static function handleJsonError($lastError)
{
    if ($lastError === JSON_ERROR_NONE) {
        return;
    }

    foreach (static::$jsonErrorMessages as $const => $message) {
          if (defined($const) && constant($const) === $lastError) {
              throw new InvalidArgumentException($message, $lastError);
          }
    }
    if (function_exists('json_last_error_msg')) {
          throw new InvalidArgumentException(json_last_error_msg(), $lastError);
    }
    throw new InvalidArgumentException('Unknown JSON encoding/decoding error.');
}
```

Reply:
@samdark ok, reverted. it's not so important.